src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -392,6 +399,101 @@ public EventBookingDTO requestBooking(final IsaacEventPageDTO event, final Regis
+            for (RegisteredUserDTO user : users) {
+                // attempt to book them on the event
+                EventBookingDTO reservation;
+
+                // attempt to book them on the event
+                if (this.hasBookingWithStatus(event.getId(), user.getId(), BookingStatus.CANCELLED)) {
+                    // if the user has previously cancelled we should let them book again.
+                    reservation = this.bookingPersistenceManager.updateBookingStatus(event.getId(), user.getId(), reservingUser.getId(),
+                            BookingStatus.RESERVED, null);
+                } else {
+                    reservation = this.bookingPersistenceManager.createBooking(event.getId(), user.getId(), reservingUser.getId(),
+                            BookingStatus.RESERVED, null);
+                }
+                reservations.add(reservation);
+
+                try {
+                    // af599 TODO: Use the correct email template here.
+                    // af599 TODO: Content team action.
+                    emailManager.sendTemplatedEmailToUser(user,
+                            emailManager.getEmailTemplateDTO("email-event-booking-confirmed"),
+                            new ImmutableMap.Builder<String, Object>()
+                                    .put("contactUsURL", generateEventContactUsURL(event))
+                                    .put("authorizationLink", String.format("https://%s/account?authToken=%s",
+                                            propertiesLoader.getProperty(HOST_NAME), event.getIsaacGroupToken()))
+                                    .put("event.emailEventDetails", event.getEmailEventDetails() == null ? "" : event.getEmailEventDetails())
+                                    .put("event", event)
+                                    .build(),
+                            EmailType.SYSTEM,
+                            Collections.singletonList(generateEventICSFile(event, reservation)));
+
+                } catch (SegueDatabaseException | ContentManagerException e) {
+                    log.error(String.format("Unable to send welcome email (%s) to user (%s)",
+                            event.getId(), user.getEmail()), e);
+                }
+
+                // auto add them to the group and grant the owner permission
+                if (event.getIsaacGroupToken() != null && !event.getIsaacGroupToken().isEmpty()) {
+                    try {
+                        this.userAssociationManager.createAssociationWithToken(event.getIsaacGroupToken(), user);
+                    } catch (InvalidUserAssociationTokenException e) {
+                        log.error(String.format("Unable to auto add user (%s) using token (%s) as the token is invalid.",
+                                user.getEmail(), event.getIsaacGroupToken()));
+                    }
+                }
```

Comment:
Not sure about the auto granting of user association permissions for reservations when this could bypass the philosophy of students always having to positively grant access to users. Is it possible that this endpoint could be misused?

Reply:
I'll file this under "it was already there", as in fact it is already there in many other places. It did struck me as odd, but I didn't have time to question existing code :)

Reply:
I think this should only be done when the student confirms their place on the event, not when the teacher reserves a space for their student on the event.
On the booking form there is bit of text explaining that they give permission to the event managers to see their work and relevant details, they will need to go through the booking form to promote from reserved to confirmed.

Reply:
Fixed in 4cb805b.